### DISCUSSION TITLE
APIConnectionError: Connection error when query endpoint via OpenAI SDK

### DISCUSSION THREAD

**Discussion by jessiewen-databricks** [stikkireddy/mlflow-extensions#54] [Q&A]
The databricks model serving endpoint is up and can be queried, but getting connection error with OpenAI SDK
<img width="738" alt="image" src="https://github.com/user-attachments/assets/46836c5a-34f2-481b-bdb2-a190f8798ce7">

APIConnectionError: Connection error.
File /local_disk0/.ephemeral_nfs/envs/pythonEnv-eaea0e34-cd98-4377-9cdc-af8c24253657/lib/python3.10/site-packages/mlflow_extensions/serving/compat/__init__.py:79, in BaseCustomMLFlowHttpClient._process_response(response_json, orig_request)
     78 try:
---> 79     prediction = response_json["predictions"][0]
     80 except KeyError:
File /local_disk0/.ephemeral_nfs/envs/pythonEnv-eaea0e34-cd98-4377-9cdc-af8c24253657/lib/python3.10/site-packages/openai/_base_client.py:1007, in SyncAPIClient._request(self, cast_to, options, remaining_retries, stream, stream_cls)
    997         return self._retry_request(
    998             input_options,
    999             cast_to,
   (...)
   1003             response_headers=None,
   1004         )
   1006     log.debug("Raising connection error")
-> 1007     raise APIConnectionError(request=request) from err
   1009 log.debug(
   1010     'HTTP Response: %s %s "%i %s" %s',
   1011     request.method,
   (...)

**Comment by stikkireddy** [comment]
https://github.com/stikkireddy/mlflow-extensions/pull/55/files

this PR will update the error messages to be better. And provide you a healthcheck option to ensure the model is still up and running or being restarted due to running out of memory.
